libtcmu.c

```diff
@@ -506,9 +516,11 @@ static int add_device(struct tcmulib_context *ctx, char *dev_name,
 		/*
 		 * Force a retry of the outstanding commands.
 		 */
-		ret = tcmu_exec_cfgfs_dev_action(dev, "reset_ring", 1);
-		if (ret)
-			tcmu_dev_err(dev, "Could not reset ring %d.\n", ret);
+		if (tcmu_is_ring_reset_support(dev)) {
+			ret = tcmu_exec_cfgfs_dev_action(dev, "reset_ring", 1);
+			if (ret)
+				tcmu_dev_err(dev, "Could not reset ring %d.\n", ret);
+		}
```

Comment:
But here would just fail the path.

Reply:
I did not understand this comment.

Reply:
I meant without this changes here, if the reset ring is not support, here will return an error which will fail the current operation. I think we should also just output one WARN message and skip this setting.